Form/Type/MultiUploadType.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\MediaBundle\Form\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Maximilian Behrsing <m.behrsing@gmail.com>
+ */
+class MultiUploadType extends AbstractType
+{
+    /**
+     * @var string
+     */
+    protected $class;
+
+    /**
+     * @param string $class
+     */
+    public function __construct($class)
+    {
+        $this->class = $class;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        // NEXT_MAJOR: Remove BC trick when bumping Symfony requirement to 2.8+
+        $hiddenType = method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')
+            ? 'Symfony\Component\Form\Extension\Core\Type\HiddenType'
+            : 'hidden';
+
+        $builder
+            ->add('context', $hiddenType, array('data' => $options['context']))
+            ->add('providerName', $hiddenType, array('data' => $options['provider']))
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver
+            ->setDefaults(
+                array(
+                    'data_class' => $this->class,
+                    'provider' => '',
+                    'context' => 'default',
+                )
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBlockPrefix()
+    {
+        return 'sonata_media_multi_upload_type';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
```

Comment:
It does on 2.3 : https://github.com/symfony/symfony/blob/2.3/src/Symfony/Component/Form/Extension/Core/Type/TextType.php#L32